src/timeline/Timeline.js

```diff
@@ -27,26 +28,47 @@ function range(from, to) {
 let {width: dimensionWidth} = Dimensions.get('window');
 
 export default class Timeline extends React.PureComponent {
+  static propTypes = {
+    start: PropTypes.number,
+    end: PropTypes.number,
+    eventTapped: PropTypes.func,
+    format24h: PropTypes.bool,
+    events: PropTypes.arrayOf(PropTypes.shape({
+      start: PropTypes.string.isRequired,
+      end: PropTypes.string.isRequired,
+      title: PropTypes.string.isRequired,
+      summary: PropTypes.string.isRequired,
+      color: PropTypes.string
+    })).isRequired
+  }
+
+  static defaultProps = {
+    start: 0,
+    end: 24,
+    events: [],
+    format24h: true
+  }
+
   constructor(props) {
     super(props);
-    const {start = 0, end = 24} = this.props;
-    this.styles = styleConstructor(props.styles, (end - start) * 100);
+    const {start, end} = this.props;
     this.calendarHeight = (end - start) * 100;
+    this.styles = styleConstructor(props.styles, this.calendarHeight);
     const width = dimensionWidth - LEFT_MARGIN;
     const packedEvents = populateEvents(props.events, width, start);
     let initPosition =
       _.min(_.map(packedEvents, 'top')) - this.calendarHeight / (end - start);
-    initPosition = initPosition < 0 ? 0 : initPosition;
+    const verifiedInitPosition = initPosition < 0 ? 0 : initPosition;
     this.state = {
-      _scrollY: initPosition,
-      packedEvents,
+      _scrollY: verifiedInitPosition,
+      packedEvents
     };
   }
 
-  componentWillReceiveProps({events, start = 0}) {
+  UNSAFE_componentWillReceiveProps({events, start = 0}) {
```

Comment:
Can we please use another lifecycle for this logic? which isn't deprecated?

Reply:
Fixed this by changing to componentDidUpdate.